### PR TITLE
Add Engineering Team Management

### PR DESCRIPTION
**Pull request by kdeldycke**
**https://github.com/kdeldycke/awesome-management**

**Why is this list awesome?**

You wonder what it looks like to be a manager from the inside? You just started your first job as the leader of a team? You're stuck into the day-to-day operations of the job? You need to move up to the next level? This awesome list is written for you!

This body of work stands out from generic leadership and management literature, by providing uncompromising insights and practical advices to bootstrap your journey into the management career track, from a technical background.

This list provides a progression to help with the transition to management. From general to specifics. It starts with an overview of the role, then describes its requirements, its position relatives to others. Then we delves into the day-to-day tools of the trade, both organizational and behavioral. At last we discuss about some of the dark sides of the job.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

❗I read this PR template multiple times. Better yet: I contributed to refining it: https://github.com/sindresorhus/awesome/commit/488c1051a3089e386ed223a5b45ab8ff8c53397c

## Requirements for your pull request

- [x] **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- [x] **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
  Sort of. I'm involved in the community by other means:
    ❗ I already contributed to this repository: https://github.com/sindresorhus/awesome/commits?author=kdeldycke
    ❗ I pushed code to `awesome-lint` to automate even more the work of reviewing lists: https://github.com/sindresorhus/awesome-lint/commits?author=kdeldycke
- [x] You have read and understood the [instructions for creating a list](create-list.md).
- [x] This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- [x] Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- [x] Your entry should be added at the bottom of the appropriate category.
- [x] The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- [x] **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
  ❗This repository has been active since January 2019: https://github.com/kdeldycke/awesome-management/commit/6d161db7ae1b1295736fbe01988e48dace111942
- [x] Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- [x] Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
  ❗There is false positives with `awesome-lint`, which I reported and proposed a PR at: https://github.com/sindresorhus/awesome-lint/pull/127 
- [x] The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- [x] **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- [x] It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
  ❗I worked on this for 2 years, made 500+ commits, highly curated the list (we're far from a simple collection of links), and was involved into the community
- [x] The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- [x] The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- [x] Non-generated Markdown file in a GitHub repo.
- [x] The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- [x] Not a duplicate. Please search for existing submissions.
- [x] Only has awesome items. Awesome lists are curations of the best, not everything.
- [x] Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- [x] Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- [x] Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- [x] Includes the [Awesome badge](awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- [x] Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- [x] Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=master&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- [x] Has [contribution guidelines](awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- [x] All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- [x] Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- [x] Doesn't use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- [x] Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- [x] Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

### CODE REVIEW

**Comment by kdeldycke** [comment]
🦄

File: readme.md [hunk 1]
```diff
@@ -540,6 +540,7 @@
 - [Clean Tech](https://github.com/nglgzz/awesome-clean-tech#readme) - Fighting climate change with technology.
 - [Wardley Maps](https://github.com/wardley-maps-community/awesome-wardley-maps#readme) - Provides high situational awareness to help improve strategic planning and decision making.
 - [Social Enterprise](https://github.com/RayBB/awesome-social-enterprise#readme) - Building an organization primarily focused on social impact that is at least partially self-funded.
+- [Engineering Team Management](https://github.com/kdeldycke/awesome-management) - How to transition from software development to engineering management, and keep up with the day-to-day, including the dark side of the job.
```
Comment (sindresorhus) [inline readme.md:543]:
```suggestion
- [Engineering Team Management](https://github.com/kdeldycke/awesome-management#readme) - How to transition from software development to engineering management.
```

**Comment by sindresorhus** [comment]
Run https://github.com/sindresorhus/awesome-lint and fix issues.

**Comment by sindresorhus** [comment]
I would remove the text title as the header image already contains that. Now it looks weird being duplicated.

**Comment by kdeldycke** [comment]
Thanks @sindresorhus for the quick review!

- [x] I merged the simplification of the entry's description as you suggested at https://github.com/sindresorhus/awesome/pull/1887#discussion_r511624586

- [x] `awesome-lint` reports several issues:

    ```shell-session
    ❯ awesome-lint --version
    0.16.0
    ```

    ```shell-session
    ❯ awesome-lint ./readme.md
    ✖ Linting

      readme.md:987:3
      ✖     1:1    Missing main list heading                           remark-lint:awesome-heading
      ✖   133:176  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   205:108  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   217:115  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   221:98   List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   223:178  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   307:104  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   343:151  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   357:147  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   413:57   List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   423:360  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   437:82   List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   601:65   List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   777:124  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   949:163  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   965:145  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖   987:3    https://danluu.com/wat/                             remark-lint:double-link
      ✖   997:105  List item description must start with valid casing  remark-lint:awesome-list-item
      ✖  1003:308  https://danluu.com/wat/#fn:P                        remark-lint:double-link
    
      19 errors
    ```

    - The first one is because I got rid of the top title header in https://github.com/kdeldycke/awesome-management/commit/29e3acdeb0be8ca90657805496db6ddd6ab6d02c as [you suggested](https://github.com/sindresorhus/awesome/pull/1887#issuecomment-716181630).

    - The `List item description must start with valid casing` / `remark-lint:awesome-list-item` issues will be fixed by this PR in `awesome-lint`: https://github.com/sindresorhus/awesome-lint/pull/127

    - The `remark-lint:double-link` issues seems to be a bug already reported at: https://github.com/sindresorhus/awesome-lint/issues/114 and https://github.com/sindresorhus/awesome-lint/issues/122

    Should we considers these issues as false-positive?

**Comment by wernerkrauss** [comment]
You can tell awesome lint to ignore some rules, e.g. the missing header.

I also noticed, that your repository doesn't match your list name.

Shouldn't the contributing guidelines be in the project root? Why are they in the hidden folder ".github"?

**Comment by kdeldycke** [comment]
> You can tell awesome lint to ignore some rules, e.g. the missing header.

You're right, I forgot about that. Thanks for the reminder! Done in: https://github.com/kdeldycke/awesome-management/commit/faf71a648fb3ee8a8ad0c7ce4ae816617ac4b97e

> I also noticed, that your repository doesn't match your list name.

Yes I had a hard time deciding between the two.

* `Awesome Management` as a title is too broad. I don't pretend to answer all management questions with my list, but only for the industry in which I practiced it. And for which I can bring some experience and critic. Management is diverse and is highly domain-dependent. If you can share most of the thinking framework and some concepts between activities, you do not have the same kind of rituals in say, a basketball team and a team of engineers. This awesome list focus on the later. That's why it is titled `Awesome Engineering Team Management`

* I created the repository with `awesome-management` name 2 years ago without any plan. It's only later, with constant curation that it became a bigger project, focused on the `Engineering Team` part, as I grew with the job. I kept it that way as it was convenient and short. I have no issue renaming it `awesome-engineering-team-management` but I'm afraid it look too noisy. What's your final opinion? Maybe we can have @sindresorhus' comment here! 🙂 

> Shouldn't the contributing guidelines be in the project root? Why are they in the hidden folder ".github"?

That is [allowed by GitHub itself](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/setting-guidelines-for-repository-contributors) to keep the root directory clean and tidy. And it is enforced by `awesome-lint` since: https://github.com/sindresorhus/awesome-lint/pull/102

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1321821945160192000

**Comment by sindresorhus** [comment]
> I created the repository with awesome-management name 2 years ago without any plan. It's only later, with constant curation that it became a bigger project, focused on the Engineering Team part, as I grew with the job. I kept it that way as it was convenient and short. I have no issue renaming it awesome-engineering-team-management but I'm afraid it look too noisy. What's your final opinion? Maybe we can have @sindresorhus' comment here! 🙂

I would personally have gone with the verbose repo name, but I don't feel strongly about it.

**Comment by kdeldycke** [comment]
Thanks a lot @sindresorhus for the merge! And long live to awesome project! ❤️ 🎉 🥳

**Comment by kdeldycke** [comment]
> I would personally have gone with the verbose repo name, but I don't feel strongly about it.

I can still change it if you like, as long as you don't mind another PR. Still have to check if GitHub properly handle redirects in the interim though.

**Comment by kdeldycke** [comment]
I just renamed the repository and the fix is available at: https://github.com/sindresorhus/awesome/pull/1889

PS: I messed up the renaming and had to recreate the repository but no worries, everything is back to normal, I just lost my 300+ GitHub stars in the process 😅.

**Comment by wernerkrauss** [comment]
OMG, I also renamed my repo to fit the renamed list and everything went fine. Didn't want you to loose your stars!

BTW: could you do me a favour and review https://github.com/sindresorhus/awesome/pull/1879 ? Maybe I miss something or am totally blind

**Comment by sindresorhus** [comment]
How is it possible to mess up renaming? Just curious.

**Comment by kdeldycke** [comment]
Once this branch got merged, I tried to clean things up and was looking to remove the `kdeldycke/awesome` repository (i.e. my fork of `sindresorhus/awesome`). I mixed things up and remove `kdeldycke/awesome-management` instead... 🤦‍♂️

**Comment by sindresorhus** [comment]
Ok. Yeah, that sucks.

**Pull request merged**